src/typechecker/FStar.TypeChecker.TcTerm.fs

```diff
@@ -83,6 +83,21 @@ let check_no_escape head_opt env (fvs:list<bv>) kt : term * guard_t =
          end in
     aux false kt
 
+let check_expected_aqual_for_binder aq b pos =
+    let expected_aq = U.aqual_of_binder b in
+    match aq, expected_aq with
+    | None, None -> aq
+    | None, Some eaq ->
+      if eaq.aqual_implicit //programmer should have written #
+      then raise_error (Errors.Fatal_InconsistentImplicitQualifier, "Inconsistent implicit qualifiers") pos
+      else expected_aq //keep the attributes
+    | Some aq, None ->
+      raise_error (Errors.Fatal_InconsistentImplicitQualifier, "Inconsistent implicit qualifiers") pos
+    | Some aq, Some eaq ->
```

Comment:
We are not checking attributes in this case, which is fine, but I guess the invariant is that aq.attrs is either empty or equal to eaq.attrs?

Reply:
This is used to check an application. 

Given `val f (#[@@@ att] x:t) : t'`

the user is expected to write `f #a` to apply `f`, matching the implicit qualifier at the binding site.
However, they do not (and cannot, there's no syntax for it) provide the attributes of the binding site at the application site.

So, this function checks that the implicit flags match and takes the attributes from the binding site, i.e., expected_aq.

I'll add this remark as a comment on the function.